src/utils/api/MainApi.js

```diff
@@ -0,0 +1,34 @@
+import { headers, setHeaders } from "../constants/constants";
+import { checkResponse } from "../callbacks/callbacks";
+
+export const { REACT_APP_BASE_URL_DEV, REACT_APP_BASE_URL_PRO } = process.env;
+const appMode = process.env.NODE_ENV
+const baseUrl = 
+  appMode === "development"
+    ? REACT_APP_BASE_URL_DEV
+    : REACT_APP_BASE_URL_PRO;
+
+export const addArticle = (articleObj, token) => {
+  setHeaders(token);
+  return fetch(`${baseUrl}/articles`, {
+    method: "POST",
+    headers: { ...headers },
+    body: JSON.stringify(articleObj),
+  }).then((res) => checkResponse(res));
```

Comment:
**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(checkResponse)
  }

```

And now you can safely replace `fetch` with `request`. And you don’t need to change anything else.